subsys/partition_manager/pm.yml.pcd

```diff
@@ -1,8 +1,10 @@
 #include <autoconf.h>
-
 # This block of RAM is used for communicating Network Core firmware update
 # metadata
 pcd_sram:
-  placement: {before: [rpmsg_nrf53_sram, end]}
+  placement: {after: [start]}
+#if defined(CONFIG_IS_PART_OF_TRUSTZONE_BUILD) || defined(CONFIG_ARM_NONSECURE_FIRMWARE)
```

Comment:
```suggestion
#if defined(CONFIG_IS_PART_OF_TRUSTZONE_BUILD)
```

pcd is only used by MCUBoot, right? I don't think MCUBoot can ever be non-secure.

Reply:
`CONFIG_ARM_NONSECURE_FIRMWARE` is for the `pcd` test, but I think you are right since it does not make since. It's not using the non-secure board so I guess I can ignore this case. I'll remove it after the CI run

Reply:
This line also looks suspicious to me.
`CONFIG_TRUSTED_EXECUTION_SECURE` and `CONFIG_ARM_SECURE_FIRMWARE`, are pretty much synonymous, was `CONFIG_ARM_SECURE_FIRMWARE` meant here?

Why exactly was the change meant away from CONFIG_TRUSTED_EXECUTION_SECURE to adding CONFIG_IS_PART_OF_TRUSTZONE_BUILD and enabling it instead?

Reply:
We had problems with CONFIG_TRUSTED_EXECUTION_SECURE because it unintetionally added pm.yml.trustzone to the build twice. Once from MCUBoot and once from (tf-m/zephyr can't remember).

Adding it twice caused trouble because pm.yml.trustzone was declaring conflicting definitions of the same partitions and it was difficult to make align the definitions due to information either missing in the zephyr build, or information missing in the mcuboot build.

I guess we could have done if(NOT DEFINED CONFIG_IS_MCUBOOT) add_trustzone_pm_yml() endif() to avoid adding it twice, but introducing a new Kconfig option seemed just as reasonable.

Reply:
Fixed.